caffe2/core/c10_operator.h

```diff
@@ -149,72 +160,47 @@ inline c10::FunctionSchema make_function_schema_for_c10(const char* OperatorName
   }
 
 // TODO This macro should take a JIT schema string instead of a vector of inputs and outputs.
-#define C10_REGISTER_CAFFE2_OPERATOR_CPU(                                      \
-    OperatorName, Inputs, Outputs, OperatorClass)                              \
-  /* Register the op schema with the c10 dispatcher */                         \
-  namespace caffe2 {                                                           \
-  namespace _c10_ops {                                                         \
-  C10_DEFINE_OP_SCHEMA(                                                        \
-      OperatorName,                                                            \
-      caffe2::detail::make_function_schema_for_c10(                            \
-          #OperatorName,                                                       \
-          Inputs,                                                              \
-          Outputs));                                                           \
-  }                                                                            \
-  /* Store the c10 operator handle so call_caffe2_op_from_c10 can access it */ \
-  namespace detail {                                                           \
-  template <>                                                                  \
-  const c10::OperatorHandle& c10_op_handle_for_c2_op<OperatorClass>() {        \
-    return caffe2::_c10_ops::OperatorName();                                   \
-  }                                                                            \
-  }                                                                            \
-  }                                                                            \
-  /* Register call_caffe2_op_from_c10 as a kernel with the c10 dispatcher */   \
-  namespace c10 {                                                              \
-  C10_REGISTER_KERNEL(caffe2::_c10_ops::OperatorName) /*.withCache<Cache>()*/  \
-      .kernel<&caffe2::detail::call_caffe2_op_from_c10<                        \
-          OperatorClass,                                                       \
-          at::DeviceType::CPU>>()                                              \
-      .dispatchKey(CPUTensorId());                                             \
+#define C10_REGISTER_CAFFE2_OPERATOR_CPU(                                     \
```

Comment:
I guess this is from the code before, but it's kind of weird that "register" also sets up the dispatch in the first place

Reply:
can you elaborate? Would you rather have two macros for setting up dispatch and registering the kernel? I think the consensus for native c10 operators was to combine it into one macro doing both, so it probably makes sense here too.